### PR TITLE
Poc/setup game

### PR DESCRIPTION
**Pull request by angelmiguelmontalvo**

### CODE REVIEW

**Comment by lufercc** [comment]
@angelmiguelmontalvo you trying to push into Master.

File: finalProject-2020-q3/code/King.cs [hunk 1]
```diff
@@ -36,7 +38,9 @@ public override CellList CaptureFreeCells(Cell[,] piecesOnBoard, int row, int co
         public override CellList AttackMovements(Cell[,] piecesOnBoard, int row, int column)
         {
             CellList cellList = ValidMovements(piecesOnBoard, row, column);
-            return (CellList)cellList.Where(cell => cell.IsEmpty() == false).ToList();
+            CellList resultList = new CellList();
+            resultList.SetList(cellList.Where(cell => cell.IsEmpty() == false).ToList());
```
Comment (jccolina) [inline finalProject-2020-q3/code/King.cs]: @angelmiguelmontalvo I think is necessary to add the Color validation "&& cell.piece.Color != Color" to line 42

File: finalProject-2020-q3/code/Knight.cs [hunk 2]
```diff
@@ -11,8 +11,10 @@ class Knight : Piece
 
         public override CellList AttackMovements(Cell[,] piecesOnBoard, int row, int column)
         {
-            CellList cellList = ValidMovements(piecesOnBoard, row, column);
-            return (CellList)cellList.Where(cell => cell.IsEmpty() == false).ToList();
+            CellList cellList = ValidMovements(piecesOnBoard, row, column); 
+            CellList resultList = new CellList();
+            resultList.SetList(cellList.Where(cell => cell.IsEmpty() == false).ToList());
```
Comment (jccolina) [inline finalProject-2020-q3/code/Knight.cs]: @angelmiguelmontalvo I think is necessary to add the Color validation "&& cell.piece.Color != Color"

File: finalProject-2020-q3/code/Rook.cs [hunk 3]
```diff
@@ -12,7 +12,9 @@ public class Rook : Piece, ICastling
         public override CellList AttackMovements(Cell[,] piecesOnBoard, int row, int column)
         {
             CellList cellList = ValidMovements(piecesOnBoard, row, column);
-            return (CellList)cellList.Where(cell => cell.IsEmpty() == false).ToList();
+            CellList resultList = new CellList();
+            resultList.SetList(cellList.Where(cell => cell.IsEmpty() == false).ToList());
```
Comment (jccolina) [inline finalProject-2020-q3/code/Rook.cs]: @angelmiguelmontalvo I think is necessary to add the Color validation "&& cell.piece.Color != Color"

File: finalProject-2020-q3/game/Game.cs [hunk 4]
```diff
@@ -34,15 +36,35 @@ public Boolean IsValidCommand(string command)
 				var target = cells[1].Value;
 				result = String.Equals(source, target, StringComparison.OrdinalIgnoreCase);
 				return !result;
-			}
+			} 
+			result = Regex.IsMatch(command, SelectRegex);
```
Comment (jccolina) [inline finalProject-2020-q3/game/Game.cs]: @angelmiguelmontalvo I think here we could re-use the IsMovement and IsSelect validation something like:
if(IsSelectCommand(command)) 
{return true}
else{
return IsMovementCommand(command);
}

**Review by jccolina — CHANGES_REQUESTED** [review]

**Review by jccolina — APPROVED** [review]

**Pull request merged**